autoPyTorch/pipeline/components/training/trainer/RowCutOutTrainer.py

```diff
@@ -40,22 +40,23 @@ def data_preparation(self, X: np.ndarray, y: np.ndarray,
         indices = self.random_state.choice(range(1, size), max(1, np.int32(size * self.patch_ratio)),
                                            replace=False)
 
-        if not isinstance(self.numerical_columns, typing.Iterable):
+        """if not isinstance(self.numerical_columns, typing.Iterable):
             raise ValueError("{} requires numerical columns information of {}"
                              "to prepare data got {}.".format(self.__class__.__name__,
                                                               typing.Iterable,
                                                               self.numerical_columns))
         numerical_indices = torch.tensor(self.numerical_columns)
         categorical_indices = torch.tensor([index for index in indices if index not in self.numerical_columns])
-
+    
         # We use an ordinal encoder on the categorical columns of tabular data
         # -1 is the conceptual equivalent to 0 in a image, that does not
         # have color as a feature and hence the network has to learn to deal
         # without this data. For numerical columns we use 0 to cutout the features
         # similar to the effect that setting 0 as a pixel value in an image.
         X[:, categorical_indices.long()] = self.CATEGORICAL_VALUE
         X[:, numerical_indices.long()] = self.NUMERICAL_VALUE
-
+        """
+        X[:, indices] = 0
```

Comment:
So all the `nan`s in categorical columns are now `0`? 
```suggestion
        # For both numerical and categorical, 0 is equivalent to NaN
        X[:, indices] = 0
```

Reply:
More like certain features for certain rows are selected to be dropped via a mask.